src/Storefront/Resources/app/storefront/src/plugin/main-menu/offcanvas-menu.plugin.js

```diff
@@ -74,9 +80,28 @@ export default class OffcanvasMenuPlugin extends Plugin {
         OffCanvas.open(this._content, this._registerEvents.bind(this), this.options.position);
         OffCanvas.setAdditionalClassName(this.options.additionalOffcanvasClass);
 
+        this._removeNotNeededBackdrops();
+
         this.$emitter.publish('openMenu');
     }
 
+    _openMenuViaUrlParameter() {
+        const urlParams = new URLSearchParams(window.location.search);
+        if (urlParams.has('offcanvas') && urlParams.get('offcanvas') === 'menu') {
+            document.querySelector('[data-off-canvas-menu="true"]')?.click();
```

Comment:
I assume we do it like this and programatically click the element because `openMenu` method has a dependency on the click event?

Reply:
We could also set the event optional. This was just the first quick draft.

Reply:
If we use `this._openMenu()` we will end-up in a loop 🤣